Bottom: 67e9c6c54cb90a8d2cbd3353e2ee6edd5c36d6d6
Top:    ed0ab8060c7f9d9196d985c17954f15cd1df4e89
Author: Duncan Mac-Vicar P <dmacvicar@suse.de>
Date:   2013-03-06 16:40:34 +0100




---

diff --git a/createrepo/__init__.py b/createrepo/__init__.py
index 69610f6..709d6a5 100644
--- a/createrepo/__init__.py
+++ b/createrepo/__init__.py
@@ -85,6 +85,7 @@ class MetaDataConfig(object):
         self.file_patterns = ['.*bin\/.*', '^\/etc\/.*', '^\/usr\/lib\/sendmail$']
         self.dir_patterns = ['.*bin\/.*', '^\/etc\/.*']
         self.skip_symlinks = False
+        self.skip_symlinked_dirs = False
         self.pkglist = []
         self.database_only = False
         self.primaryfile = 'primary.xml.gz'
@@ -281,6 +282,8 @@ class MetaDataGenerator:
         func(arg, top, names)
         for name in names:
             name = os.path.join(top, name)
+            if self.conf.skip_symlinked_dirs and os.path.islink(name):
+                continue
             if os.path.isdir(name):
                 self._os_path_walk(name, func, arg)
     def getFileList(self, directory, ext):
diff --git a/genpkgmetadata.py b/genpkgmetadata.py
index 8c98191..f9104e7 100755
--- a/genpkgmetadata.py
+++ b/genpkgmetadata.py
@@ -92,6 +92,8 @@ def parse_args(args, conf):
         help="<dir> = optional directory to output to")
     parser.add_option("-S", "--skip-symlinks", dest="skip_symlinks",
         default=False, action="store_true", help="ignore symlinks of packages")
+    parser.add_option("--skip-symlinked-dirs", dest="skip_symlinked_dirs",
+        default=False, action="store_true", help="ignore symlinked directories")
     parser.add_option("--changelog-limit", dest="changelog_limit",
         default=None, help="only import the last N changelog entries")
     parser.add_option("--unique-md-filenames", dest="unique_md_filenames",
